### ISSUE TITLE
[Bug]: Substitution in custom charts do not work

### ISSUE THREAD

**Issue by Nikhil-Nandagopal** [appsmithorg/appsmith#8710]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

using mustache inside the data part of a chart does not work

### Steps To Reproduce

1. Drag a chart
2. Switch to a custom chart
3. Wrap mustache around its data property and see an error

```
{
  "type": "column2d",
  "dataSource": {
    "chart": {
      "caption": "Sales Report",
      "xAxisName": "Product Line",
      "yAxisName": "Revenue($)",
      "theme": "fusion"
    },
    "data": {{[
      {
        "label": "Product1",
        "value": 20000
      },
      {
        "label": "Product2",
        "value": 22000
      },
      {
        "label": "Product3",
        "value": 32000
      }
    ]}}
  }
}
```

### Environment

Production

### Version

Cloud

**Comment by Nikhil-Nandagopal** [comment]
@hetunandu @ApekshaBhosale I think this is a problem with JS Eval right?

**Comment by hetunandu** [comment]
Fixed with #9191
